### PR TITLE
Add plo for riscv64

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

JIRA: RTOS-550
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by nalajcie** [comment]
I would advice to add common technical words to `/.codespell_ignore` file in repo to avoid future issues with codespell
image

AFAIK it's only used in `ports` right now - in lsb PR (https://github.com/phoenix-rtos/phoenix-rtos-ports/pull/54/files#diff-9419e32c35e70ac3f5a58e5d4a2312c38b9d0b4549abaa63829d9143b50408b7)

File: hal/riscv64/_init.S [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#define __ASSEMBLY__
+
+#include "cpu.h"
+#define SR_SUM          0x00040000
+#define SR_MER          0x00080000
```
Comment (agkaminski) [inline hal/riscv64/_init.S:20]:
```suggestion
#define SR_SUM 0x00040000
#define SR_MER 0x00080000
```

File: hal/riscv64/cpu.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#ifndef __ASSEMBLY__
+
+
+#define csr_set(csr, val) \
+	({ \
+		unsigned long __v = (unsigned long)(val); \
+		__asm__ __volatile__("csrs " #csr ", %0" \
+							 : \
+							 : "rK"(__v) \
+							 : "memory"); \
```
Comment (agkaminski) [inline hal/riscv64/cpu.h:39]: clang-format is getting confused by inline asm
```suggestion
		__asm__ __volatile__("csrs " #csr ", %0" :: "rK"(__v) : "memory");
```

File: hal/riscv64/timer.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#define TIMER_IRQ (5u | CLINT_IRQ_FLG)
+
+
+static struct {
+	volatile time_t jiffies;
+	const u32 interval;
+} timer_common = {
+	.jiffies = 0,
+	.interval = TIMER_FREQ / 1000, /* 1ms */
+};
```
Comment (agkaminski) [inline hal/riscv64/timer.c:28]: Will work now thanks to ld scripts, but should be in init by convention.

File: hal/riscv64/_init.S (hunk 1 above)
Comment (lukileczo) [reply hal/riscv64/_init.S]: not needed anyway, removed it

**Review by agkaminski — APPROVED** [review]

**Pull request merged**